### PR TITLE
PREOPS-858: Initial plot-navigator chart.

### PR DESCRIPTION
**Pull request by ctslater**

### CODE REVIEW

File: charts/plot-navigator/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion}}
+        imagePullPolicy: Always
+        env:
+          - name: BUTLER_URI
+            value: {{ .Values.butler_uri | quote }}
+        volumeMounts:
+          - name: butler-secret
+            mountPath: "/home/worker/.lsst/"
+            # Or mount this somewhere else and copy the file to .lsst
+            # mountPath: "/opt/lsst/software/jupyterlab/butler-secret"
```
Comment (spenczar) [inline charts/plot-navigator/templates/deployment.yaml:31]: Cruft to be deleted, maybe?

File: charts/plot-navigator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+hostname: ""
```
Comment (spenczar) [inline charts/plot-navigator/values.yaml:1]: Why would `hostname` be different from `ingress.host`?

File: charts/plot-navigator/values.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+butler_uri: ""
+
+# override in phalanx
+butler_secret_path: ""
+
+ingress:
+  enabled: true
+  host: ""
+  path: "/plot-navigator"
+  pathType: ImplementationSpecific
```
Comment (spenczar) [inline charts/plot-navigator/values.yaml:19]: Consider using [helm-docs](https://github.com/norwoodj/helm-docs) to document each of the meaning of each of these values, and then generating a README.md with helm-docs. 

For example, here's Gafaelfawr's [values.yaml](https://github.com/lsst-sqre/charts/blob/master/charts/gafaelfawr/values.yaml) and the resulting generated [README.md](https://github.com/lsst-sqre/charts/blob/master/charts/gafaelfawr/README.md).

File: charts/plot-navigator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+hostname: ""
+# Include leading forward slash in basePath
+basePath: "/plot-navigator"
```
Comment (spenczar) [inline charts/plot-navigator/values.yaml:3]: Why would `basename` be different from `ingress.path`?

File: charts/plot-navigator/templates/service.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+kind: Service
+metadata:
+  name: plot-navigator
+spec:
+  selector:
+    app: plot-navigator
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: {{ .Values.port | default 8080 }}
```
Comment (spenczar) [inline charts/plot-navigator/templates/service.yaml:12]: `port` appears to be missing from `Values.yaml`. You have a default so it probably all works, but I think this is a little less legible from the outside than including `port` with a default.

**Review by spenczar — COMMENTED** [review]
Some drive-by comments. I don't know anything about Vault secrets, so you might want some extra attention on that.

**Comment by cbanek** [comment]
Looks good, deployed it on minikube and it didn't work but I don't have butler, but it made the secret and such.

**Review by cbanek — APPROVED** [review]

File: charts/plot-navigator/values.yaml (hunk 2 above)
Comment (ctslater) [reply charts/plot-navigator/values.yaml:1]: Agreed, merged into a single parameter now.

File: charts/plot-navigator/templates/service.yaml (hunk 5 above)
Comment (ctslater) [reply charts/plot-navigator/templates/service.yaml]: I don't have a good reason for this to be configurable, so I just removed it.

**Pull request merged**